korman/operators/op_mesh.py

```diff
@@ -26,6 +26,141 @@ def poll(cls, context):
+
+        for obj in bpy.data.objects:
+            obj.select = False
+
+        # Create Empty, rotated to horizontal
+        flare_root = bpy.data.objects.new("{}".format(self.name_stem), None)
+        flare_root.empty_draw_type = "SINGLE_ARROW"
+        flare_root.matrix_world = cursor_shift
+        flare_root.rotation_euler[0] = math.radians(90)
+        flare_root.plasma_object.enabled = True
+        flare_root.select = True
+        bpy.context.scene.objects.link(flare_root)
+        bpyscene.objects.active = flare_root
+
+        # Enable VFM on Empty
+        bpy.ops.object.plasma_modifier_add(types="viewfacemod")
+        flare_root.plasma_modifiers.viewfacemod.preset_options = "Sprite"
+
+        # Create a textured Plane
+        with utils.bmesh_object("{}_Visible".format(self.name_stem)) as (flare_plane, bm):
+            flare_plane.hide_render = True
+            flare_plane.plasma_object.enabled = True
+            bpyscene.objects.active = flare_plane
+
+            # Make the actual plane mesh, facing away from the empty
+            bmesh.ops.create_grid(bm, size=(0.5 + self.flare_distance * 0.5), matrix=mathutils.Matrix.Rotation(math.radians(180.0), 4, 'X'))
+            bmesh.ops.transform(bm, matrix=mathutils.Matrix.Translation((0.0, 0.0, -self.flare_distance)), space=flare_plane.matrix_world, verts=bm.verts)
+            bpy.ops.object.origin_set(type="ORIGIN_GEOMETRY")
+
+        # Give the plane a basic UV unwrap, so that it's texture-ready
+        bpy.ops.object.editmode_toggle()
+        bpy.ops.uv.smart_project()
+        bpy.ops.object.editmode_toggle()
+
+        # Create AUTOGEN_FLARE material and texture, setting No-Z-Write
+        auto_mat = self.find_create_material()
+        flare_plane.data.materials.append(auto_mat)
+
+        # Parent Plane to Empty
+        flare_plane.parent = flare_root
+
+        # Enable Opacity Fader
+        bpy.ops.object.plasma_modifier_add(types="fademod")
+        flare_plane.plasma_modifiers.fademod.fader_type = "FadeOpacity"
+        flare_plane.plasma_modifiers.fademod.fade_in_time = 0.25
+        flare_plane.plasma_modifiers.fademod.fade_out_time = 0.25
+        flare_plane.plasma_modifiers.fademod.bounds_center = True
+
+    def find_create_material(self):
+        # If the selected flare material exists, use it
+        auto_mat = bpy.data.materials.get(self.flare_material_name, None)
+
+        if auto_mat is None:
+            # Generate a new flare material and texture
+            auto_mat = bpy.data.materials.new(name=FLARE_MATERIAL_BASE_NAME)
+            auto_mat.use_shadeless = True
+            auto_mat.use_shadows = False
+            auto_mat.use_cast_shadows = False
+            self.flare_material_name = auto_mat.name
+
+            auto_tex = bpy.data.textures.new(name=FLARE_MATERIAL_BASE_NAME, type="IMAGE")
+            auto_tex.use_alpha = True
+            auto_tex.plasma_layer.skip_depth_write = True
```

Comment:
You probably also want to set `layer_props.z_bias` and `layer_props.skip_depth_test`. These allow the flare to shine past the edges of partially occluding objects (but then the FadeOpacityMod hides them when completely occluded)

Reply:
Thanks, I'll try those out!

Reply:
Added!